snakemake/scheduler.py

```
@@ -737,7 +737,111 @@ def calc_reward():
+                # Check resources
+                exhausted_some_res = False
+                for i in range(len(max_glob_res)):
+                    # Check if exceeds max
+                    if used_res[i] + job_res[i] > max_glob_res[i]:
+                        exhausted_some_res = True
+                        break
+
+                # If limits not yet exceeded, add job.
+                if not exhausted_some_res:
+                    # Update total resources
+                    for i in range(len(res)):
```

Ruff

Undefined name `res`

(F821)